dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/OpenAPIVersionConfigImpl.java

```
@@ -0,0 +1,282 @@
+
+        String versionString = (String) properties.get(VERSION_KEY);
+
+        if (Objects.equals(versionString, configuredVersionString)) {
+            // Our config has not changed, return immediately to avoid potentially logging warnings
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Config unchanged", versionString);
+            }
+            return;
+        }
+
+        configuredVersionString = versionString;
+
+        if (versionString == null) {
+            this.configuredVersion = defaultVersion();
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "No OpenAPI version set, default version used", this.configuredVersion);
+            }
+            return;
+        }
+
+        Optional<OpenAPIVersion> version = OpenAPIVersion.parse(versionString);
+        if (!version.isPresent()) {
+            this.configuredVersion = defaultVersion();
+            Tr.warning(tc,
+                       MessageConstants.OPENAPI_VERSION_INVALID_CWWK06181W,
+                       versionString,
+                       getSupportedVersions(),
+                       this.configuredVersion);
+            return;
+        }
+
+        if (!isSupported(version.get())) {
+            this.configuredVersion = defaultVersion();
+            Tr.warning(tc,
+                       MessageConstants.OPENAPI_VERSION_NOT_SUPPORTED_CWWK06182W,
+                       version.get(),
+                       getSupportedVersions(),
+                       this.configuredVersion);
+            return;
+        }
+
+        // Configured version is acceptable
+        this.configuredVersion = version.get();
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(this, tc, "OpenAPI version set", this.configuredVersion);
+        }
+    }
+
+    @Override
+    public void applyConfig(OpenAPI model) {
```

This method does not beta guard correctly. It doesn't throw an UnsupportedOperationException if we're not beta, issue a warning message, and isn't marked deprecated.

---

Will check, I think throwing UnsupportedOperationException and marking deprectated is only needed when we're adding methods to API or SPI classes. Since the user has access to that class, they could call that method in a non-beta version, at which point we want to throw an exception.

In contrast, this method does get called for both beta and production versions, it just does nothing in the production version.

---

Those rules are indeed [only for API/SPI methods](https://github.com/OpenLiberty/open-liberty/wiki/Beta-Fencing#beta-fencing-using-jvm-system-property).

However, I did notice while checking the beta fencing that I haven't updated the metatype :facepalm:

---

I've now updated the metatype.